test/common/stats/tag_extractor_impl_test.cc

```diff
@@ -28,8 +29,35 @@ TEST(TagExtractorTest, TwoSubexpressions) {
   EXPECT_EQ("cluster_name", tags.at(0).name_);
 }
 
+TEST(TagExtractorTest, RE2Variants) {
+  re2::RE2 re("^cluster\\.((.*?)\\.)");
+  re2::StringPiece match1, match2;
+  ASSERT_TRUE(
+      re2::RE2::PartialMatch("cluster.test_cluster.upstream_cx_total", re, &match1, &match2));
+  EXPECT_EQ("test_cluster.", std::string(match1));
+  EXPECT_EQ("test_cluster", std::string(match2));
+
+  re2::RE2 alternate_re("^cluster\\.([^\\.]+)\\..*");
+  ASSERT_TRUE(re2::RE2::FullMatch("cluster.test_cluster.upstream_cx_total", alternate_re, &match1));
+  EXPECT_EQ("test_cluster", std::string(match1));
+
+  re2::RE2 alternate_re2("^cluster\\.(([^\\.]+)\\.).*");
+  ASSERT_TRUE(re2::RE2::FullMatch("cluster.test_cluster.upstream_cx_total", alternate_re2, &match1,
+                                  &match2));
+  EXPECT_EQ("test_cluster.", std::string(match1));
+  EXPECT_EQ("test_cluster", std::string(match2));
+
+  std::regex sr("^cluster\\.((.*?)\\.)");
+  std::match_results<std::string::const_iterator> smatch;
+  std::string str("cluster.test_cluster.upstream_cx_total");
+  ASSERT_TRUE(std::regex_search(str, smatch, sr));
+  ASSERT_LE(3, smatch.size());
+  EXPECT_EQ("test_cluster.", std::string(smatch[1]));
+  EXPECT_EQ("test_cluster", std::string(smatch[2]));
+}
+
```

Comment:
This looks like a test of RE2, not the RE2 tag extractor.

Reply:
Good point; I was iterating on what the RE should be functionally. I switched it to use the TagMatcher interface.